tests/integrations/basic-fail/tests/ui_tests.rs

```diff
@@ -6,10 +6,10 @@ fn main() -> ui_test::color_eyre::Result<()> {
     let mut config = Config {
         quiet: false,
         root_dir: "tests/actual_tests".into(),
-        dependencies_crate_manifest_path: Some("tests/dummy_crate/Cargo.toml".into()),
+        dependencies_crate_manifest_path: Some("Cargo.toml".into()),
         dependency_builder: Some(DependencyBuilder {
             program: std::path::PathBuf::from("cargo"),
-            args: vec![],
+            args: vec!["build".into()],
             envs: vec![("CARGO_TARGET_DIR".into(), path.into())],
```

Comment:
Should we set a suitable default dependency builder? I guess the only variable here is `path`, so either we find a way to not need that, or we add a function to compute a dependency builder given a target path.

Reply:
we can compute it automatically by implementing a `Default` impl manually and removing the `Option` around it

Reply:
> either we find a way to not need that,

it's not needed, just convenient to avoid rebuilds. I defaulted it to nothing and will keep using it to overwrite the `CARGO_TARGET_DIR`